PolyEngine/Core/Src/Vector2f.hpp

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include "Defines.hpp"
+#include "BasicMath.hpp"
+
+namespace Poly 
+{
+	/// <summary>Class representing 2D float vector.</summary>
+	class ALIGN_16 CORE_DLLEXPORT Vector2f : public BaseObject<>{
+	public:
+		static const Vector2f ZERO;
+		static const Vector2f UNIT_X;
+		static const Vector2f UNIT_Y;
+
+		/// <summary>Creates zero vector.</summary>
+		constexpr Vector2f() : X(0.f), Y(0.f) {}
+
+		/// <summary>Creates vector from float values.</summary>
+		/// <param name="x"></param>
+		/// <param name="y"></param>
+		constexpr Vector2f(float x, float y) : X(x), Y(y) {}
+
+
+		inline Vector2f(const Vector2f& rhs) : X(rhs.X), Y(rhs.Y) {}
+		inline Vector2f& operator=(const Vector2f& rhs) { X = rhs.X; Y = rhs.Y; return *this; }
+
+		/// <summary>Negation operator</summary>
+		inline Vector2f operator-() const { return Vector2f(-X, -Y); }
+
+		// Comparisons
+		bool operator==(const Vector2f& rhs) const;
+		inline bool operator!=(const Vector2f& rhs) const { return !(*this == rhs); }
+
+		// Operators with other vectors
+		Vector2f operator+(const Vector2f& rhs) const;
+		Vector2f operator-(const Vector2f& rhs) const;
+		Vector2f& operator+=(const Vector2f& rhs);
+		Vector2f& operator-=(const Vector2f& rhs);
+
+		// Operators with floats
+		Vector2f operator*(float rhs) const;
+		Vector2f operator/(float rhs) const;
+		Vector2f& operator*=(float rhs);
+		Vector2f& operator/=(float rhs);
+
+		/// <summary>Returns length of the vector.</summary>
+		/// <returns>Length of the vector.</returns>
+		float Length() const;
+
+		/// <summary>Returns square length of the vector.</summary>
+		/// <returns>Square length of the vector.</returns>
+		float LengthSquared() const;
+
+		/// <summary>Calculates dot product of two vectors.</summary>
+		/// <param name="rhs">The other vector.</param>
+		/// <returns>Dot product of the two vectors.</returns>
+		float Dot(const Vector2f& rhs) const;
+
+		/// <summary>Calculates cross product of two vectors.</summary>
+		/// <param name="rhs">The other vector (right).</param>
+		/// <returns>Cross product of the two vectors.</returns>
+		float Cross(const Vector2f& rhs) const;
+
+		/// <summary>Normalizes the vector.</summary>
+		/// <returns>Reference to itself.</returns>
+		Vector2f& Normalize();
+
+		/// <summary>Creates normalized vector.</summary>
+		/// <returns>Normalized vector.</returns>
+		Vector2f GetNormalized() const;
+
+		/// <summary>Creates Vector2i based on this vector.</summary>
+		/// <returns>Converted vector.</returns>
+		class Vector2i ToVector2i() const;
```

Comment:
Please move class declaration to the beginning of the namespace. Please also provide explicit cast operator to Vector2i.